l3kernel/l3keys.dtx

```diff
@@ -774,6 +774,36 @@
 %   is not processed in any way by \cs{keys_set:nn}.
 % \end{variable}
 %
+% \subsection{Expanding the values of keys}
+%
+% To allow the user to apply expansion of values when the key is set, rather
+% than when they are used, key names can be
+% followed by an expansion specifier. This is given by appending |:| and a
+% single letter specifier to the key name. The latter are the normal argument
+% specifiers for \pkg{expl3}, thus may be one of \texttt{n} (redundant but
+% supported), \texttt{o}, \texttt{V} or \texttt{e}. Thus for example
+% \begin{verbatim}
+%   \keys_define:n { mymodule } { key .code:n = \tl_show:n { "#1" } }
+%   \tl_set:Nn \l_mymodule_value_tl { value }
+%   \dime_set:N \l__mymodule_value_dim { 123pt }
+%   \keys_set:nn { mymodule } { key = value }
+%   \keys_set:nn { mymodule } { key : o = \l_mymodule_value_tl }
```

Comment:
I think here it would be better to show no space around the `:` spaces work here but just because of L3 syntax, you could equally spell the key name as `k e y` with spaces.  Unlike defining the keys, setting them is often, perhaps usually, lifted up to a document level command and there you need `key:e` with no space (unless we want to take the cost of removing spaces by hand.)

```
\documentclass{article}

\begin{document}

\ExplSyntaxOn
\keys_define:nn {me} {
  k .tl_set:N  = \l_me_foo_tl
}

\def\myset{\keys_set:nn {me}}
\def\myshow{\show\l_me_foo_tl}
\ExplSyntaxOff

\def\qq{88}

%\myset{k :e=\qq} %expansion ' e' unknown
%\myset{k :e=\qq} %key 'me/k' unknown
%\myset{k : e=\qq} %expansion ' e' unknown
\myset{k:e=\qq} % works
\myshow

\end{document}
```

Reply:
I could do that, or I could zap the spaces ... thoughts?

Reply:
I've changed the docs; we do have 'trim spaces only one side' functions, so if we wish to revisit later, the cost of space trimming isn't a disaster.